stl/inc/memory

```diff
@@ -1504,6 +1550,233 @@ private:
+public:
+    explicit _Ref_count_unbounded_array(const size_t _Count) : _Ref_count_base(), _Size(_Count) {
+        _Uninitialized_value_construct_multidimensional_n(_Get_ptr(), _Size);
+    }
+
+    explicit _Ref_count_unbounded_array(const size_t _Count, const remove_extent_t<_Ty>& _Val)
+        : _Ref_count_base(), _Size(_Count) {
+        _Uninitialized_fill_multidimensional_n(_Get_ptr(), _Size, _Val);
+    }
+
+    ~_Ref_count_unbounded_array() = default; // nothing to do, _Storage was already destroyed in _Destroy
+
+    auto _Get_ptr() {
+        return reinterpret_cast<remove_extent_t<_Ty>*>(
+            reinterpret_cast<char*>(this) + _Div_ceil(sizeof(*this), alignof(_Ty)) * alignof(_Ty));
+    }
+
+private:
+    size_t _Size;
+
+    virtual void _Destroy() noexcept override { // destroy managed resource
+        _Reverse_destroy_multidimensional_n(_Get_ptr(), _Size);
+    }
+
+    virtual void _Delete_this() noexcept override { // destroy self
+        using _Storage = aligned_storage_t<_Max_value(alignof(_Ty), alignof(_Ref_count_unbounded_array<_Ty>)),
+            _Max_value(alignof(_Ty), alignof(_Ref_count_unbounded_array<_Ty>))>;
+        this->~_Ref_count_unbounded_array();
+        delete[] reinterpret_cast<_Storage*>(this);
+    }
+};
+
+template <class _Ty>
+class _Ref_count_unbounded_array<_Ty, true> : public _Ref_count_base { // handle reference counting for unbounded array
+                                                                       // with trivial destruction in control block,
+                                                                       // no allocator
+public:
+    explicit _Ref_count_unbounded_array(const size_t _Count) : _Ref_count_base() {
+        _Uninitialized_value_construct_multidimensional_n(_Get_ptr(), _Count);
+    }
+
+    explicit _Ref_count_unbounded_array(const size_t _Count, const remove_extent_t<_Ty>& _Val) : _Ref_count_base() {
+        _Uninitialized_fill_multidimensional_n(_Get_ptr(), _Count, _Val);
+    }
+
+    ~_Ref_count_unbounded_array() = default; // nothing to do, _Ty is trivially destructible
+
+    auto _Get_ptr() {
+        return reinterpret_cast<remove_extent_t<_Ty>*>(
+            reinterpret_cast<char*>(this) + _Div_ceil(sizeof(*this), alignof(_Ty)) * alignof(_Ty));
+    }
+
+private:
+    virtual void _Destroy() noexcept override { // destroy managed resource
+        // nothing to do, _Ty is trivially destructible
+    }
+
+    virtual void _Delete_this() noexcept override { // destroy self
+        using _Storage = aligned_storage_t<_Max_value(alignof(_Ty), alignof(_Ref_count_unbounded_array<_Ty>)),
+            _Max_value(alignof(_Ty), alignof(_Ref_count_unbounded_array<_Ty>))>;
+        this->~_Ref_count_unbounded_array();
+        delete[] reinterpret_cast<_Storage*>(this);
+    }
+};
+
+// CLASS TEMPLATE _Ref_count_bounded_array
+template <class _Ty>
+class _Ref_count_bounded_array

```

Comment:
That line wrap is different than all other classes

Reply:
Fixed